aten/src/ATen/cuda/CUDAEvent.h

```diff
@@ -30,6 +31,21 @@ struct AT_CUDA_API CUDAEvent {
   CUDAEvent(unsigned int flags = DEFAULT_FLAGS)
   : flags_{flags} { }
 
+  CUDAEvent(const cudaIpcEventHandle_t* handle)
```

Comment:
I'm ambivalent about this constructor. I think just keeping the one below which is more explicit about the device is better.

If you do keep this one, it should be marked `explicit` to avoid allowing implicit conversions in C++. The above `CUDAEvent(unsigned int flags)` should also be marked explicit.